src/Compilers/VisualBasic/Portable/Semantics/Conversions.vb

```diff
@@ -2053,7 +2077,16 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
             If sourceType Is Nothing Then
                 source = source.GetMostEnclosedParenthesizedExpression()
-                sourceType = If(source.Kind <> BoundKind.ArrayLiteral, source.Type, New ArrayLiteralTypeSymbol(DirectCast(source, BoundArrayLiteral)))
+                If source.Kind = BoundKind.ArrayLiteral Then
+                    sourceType = New ArrayLiteralTypeSymbol(DirectCast(source, BoundArrayLiteral))
```

Comment:
I know this is pre-existing code, but I'm surprised that a new type needs to be constructed here. Why not do `sourceType = DirectCast(source, BoundArrayLiteral).InferredType;`?